README.md

```diff
@@ -209,8 +214,8 @@ if err != nil {
 ### Presence on a channel
 
 ```go
-page, err := channel.Presence.Get(nil)
-for ; err == nil; page, err = page.Next() {
+page, err = channel.Presence.Get(nil)
+for ; err == nil && page != nil; page, err = page.Next() {
```

Comment:
This is the first time I've seen how this API is used, so perhaps this isn't the correct place to discuss, but I'll put it here anyway since I've seen it.

The usage feels a little awkward, I wonder if something like the following would be better:

```go
params := &ably.PaginateParams{
	Limit: 10,
}
err := channel.Presence.Get(params, func(page *ably.PaginatedResult) bool {
	// iterate page.PresenceMessages()
	// return false to stop iterating
})
```

and similarly for the other paginated operations.

Reply:
Note this API isn't covered by the 1.2 preview subset and is likely to change. I just fixed it here (previous version returned an error on end of iteration, so this example would always panic).

I'm moving this discussion at https://github.com/ably/ably-go/issues/197#issuecomment-718446264 if that's OK.

Reply:
That's fine, thanks 👍